api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/payments/PaymentCalculator.java

```diff
@@ -19,33 +28,37 @@ public PaymentCalculator(@Value("${payment-cycle.maximum-balance-period}") int m
      * Returns a partial amount when the full entitlement amount plus the card balance is greater than or equal to the maximum card balance.
      * Returns zero when the card balance is greater than or equal to the maximum card balance.
      *
-     * @param entitlement a payment cycle voucher entitlement
+     * @param entitlement        a payment cycle voucher entitlement
      * @param cardBalanceInPence the card balance in pence
-     * @return the amount in pence to apply to the next payment request
+     * @return the payment calculation which comprises the amount in pence to apply to the next payment request and the status of the PaymentCycle
      */
-    public int calculatePaymentCycleAmountInPence(PaymentCycleVoucherEntitlement entitlement, int cardBalanceInPence) {
+    public PaymentCalculation calculatePaymentCycleAmountInPence(PaymentCycleVoucherEntitlement entitlement, int cardBalanceInPence) {
         int firstWeekEntitlementInPence = entitlement.getFirstVoucherEntitlementForCycle().getTotalVoucherValueInPence();
         int maximumAllowedCardBalanceInPence = firstWeekEntitlementInPence * maximumBalancePeriod;
 
         if (isCardBalanceTooHigh(cardBalanceInPence, maximumAllowedCardBalanceInPence)) {
-            return 0;
+            return NO_PAYMENT_CALCULATION;
         }
         int paymentCycleTotalEntitlementInPence = entitlement.getTotalVoucherValueInPence();
-        if (fullPaymentKeepsBalanceUnderThreshold(cardBalanceInPence, maximumAllowedCardBalanceInPence, paymentCycleTotalEntitlementInPence)) {
-            return paymentCycleTotalEntitlementInPence;
+        if (fullPaymentKeepsBalanceLessThanOrEqualToThreshold(cardBalanceInPence, maximumAllowedCardBalanceInPence, paymentCycleTotalEntitlementInPence)) {
```

Comment:
This is getting to the point where `(cardBalanceInPence + paymentCycleTotalEntitlementInPence <= maxAllowedCardBalanceInPence)` is shorter and more meaningful than the extracted method.

Reply:
After discussions, will rename to `fullPaymentKeepsBalanceWithinThreshold`.